x/datadeal/keeper/grpc_query_deal.go

```diff
@@ -24,17 +26,79 @@ func (k Keeper) Deal(goCtx context.Context, req *types.QueryDealRequest) (*types
 	}, nil
 }
 
-func (k Keeper) Deals(ctx context.Context, request *types.QueryDealsRequest) (*types.QueryDealsResponse, error) {
-	//TODO implement me
-	panic("implement me")
+func (k Keeper) Deals(goCtx context.Context, req *types.QueryDealsRequest) (*types.QueryDealsResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "invalid request")
+	}
+
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	store := ctx.KVStore(k.storeKey)
+	dealsStore := prefix.NewStore(store, types.DealKey)
+
+	var deals []*types.Deal
+	pageRes, err := query.Paginate(dealsStore, req.Pagination, func(_ []byte, value []byte) error {
+		var deal types.Deal
+		err := k.cdc.UnmarshalLengthPrefixed(value, &deal)
+		if err != nil {
+			return err
+		}
+		deals = append(deals, &deal)
+		return nil
+	})
+
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
+	return &types.QueryDealsResponse{
+		Deals:      deals,
+		Pagination: pageRes,
+	}, nil
 }
 
-func (k Keeper) Certificates(ctx context.Context, certificates *types.QueryCertificates) (*types.QueryCertificatesResponse, error) {
-	//TODO implement me
-	panic("implement me")
+func (k Keeper) Certificates(goCtx context.Context, req *types.QueryCertificates) (*types.QueryCertificatesResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "invalid request")
+	}
+
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	store := ctx.KVStore(k.storeKey)
+	certsStore := prefix.NewStore(store, types.CertificateKey)
+
+	var certs []*types.Certificate
+	pageRes, err := query.Paginate(certsStore, req.Pagination, func(_ []byte, value []byte) error {
+		var cert types.Certificate
+		err := k.cdc.UnmarshalLengthPrefixed(value, &cert)
+		if err != nil {
+			return err
+		}
+		certs = append(certs, &cert)
+		return nil
+	})
+
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
```

Comment:
```suggestion
	if err != nil {
		return nil, status.Error(codes.Internal, err.Error())
	}
```

Reply:
Umm I don't know what's different :)

Reply:
just removed empty line between error occurrence and check.

Reply:
Aha I got it.